src/components/accession2/EditStateModal.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect } from 'react';
+import strings from 'src/strings';
+import Button from '../common/button/Button';
+import DialogBox from '../common/DialogBox/DialogBox';
+import { Box, Grid, Typography } from '@mui/material';
+import { Icon, Select } from '@terraware/web-components';
+import { Accession2 } from 'src/api/accessions2/accession';
+import useForm from 'src/utils/useForm';
+import { updateAccession2 } from 'src/api/accessions2/accession';
+import { ACCESSION_2_STATES } from 'src/types/Accession';
+import { makeStyles } from '@mui/styles';
+
+const useStyles = makeStyles(() => ({
+  messageIcon: {
+    fill: '#3A4445',
+  },
+}));
+
+export interface EditStateDialogProps {
+  open: boolean;
+  accession?: Accession2;
+  onClose: () => void;
+  reload: () => void;
+}
+
+export default function EditLocationDialog(props: EditStateDialogProps): JSX.Element {
+  const classes = useStyles();
+  const { onClose, open, accession, reload } = props;
+  const [record, setRecord, onChange] = useForm(accession);
+
+  useEffect(() => {
+    setRecord(accession);
+  }, [accession, setRecord]);
+
+  const saveState = async () => {
+    if (accession && record) {
+      const response = await updateAccession2(record);
+      if (response.requestSucceeded) {
+        reload();
+      }
+      onClose();
+    }
+  };
+
+  return (
+    <DialogBox
+      onClose={onClose}
+      open={open}
+      title={strings.STATUS}
+      size='small'
+      middleButtons={[
+        <Button label={strings.CANCEL} type='passive' onClick={onClose} priority='secondary' key='button-1' />,
+        <Button onClick={saveState} label={strings.UPDATE} key='button-2' />,
+      ]}
+    >
+      <Grid item xs={12} textAlign='left'>
+        <Grid item xs={12}>
+          <Select
+            label={strings.STATUS}
+            placeholder={strings.SELECT}
+            options={ACCESSION_2_STATES}
```

Comment:
There's a table in the PRD explaining what edit states are valid. We may want to verify that. It's ok to leave it like this for now if we come back and follow up.
Basically, we cannot allow setting any state here, only some states.
See https://terraformation.atlassian.net/wiki/spaces/PM/pages/428114078/PRD+Seed+Management+updates+web+app#Accession-status-rules